File: server/apps/subjects/infrastructure/dependencies/subject_dependency.py
from apps.subjects.application.use_cases.get_subjects_use_case import (
    GetSubjectsUseCase,
)
from apps.subjects.application.use_cases.create_subject import (
    CreateSubjectUseCase,
)
from apps.subjects.infrastructure.persistence.repositories.mongo_subject_repository import (
    MongoSubjectRepository,
)
from apps.universities.infrastructure.persistence.repositories.mongo_university_repository import (
    MongoUniversityRepository,
)

from apps.departments.infrastructure.persistence.repositories.mongo_department_repository import (
    MongoDepartmentRepository,
)

subject_repository = MongoSubjectRepository()
university_repository = MongoUniversityRepository()

department_repository = MongoDepartmentRepository()

get_subjects_use_case = GetSubjectsUseCase(
    subject_repository=subject_repository,
)
create_subject_use_case = CreateSubjectUseCase(
    subject_repository=subject_repository,
    university_repository=university_repository,
    department_repository=department_repository,
)

File: client/src/infrastructure/dependencies/subject/subject_dependency.js
const SubjectRepositoryImpl = require("../../../infrastructure/repositories/subject/subject_repository_impl");

const GetSubjectsUseCase = require("../../../application/subjects/use_cases/get_subjects");
const CreateSubjectUseCase = require("../../../application/subjects/use_cases/create_subject");
const subjectRepository = new SubjectRepositoryImpl();

const getSubjectsUseCase = new GetSubjectsUseCase(subjectRepository);
const createSubjectUseCase = new CreateSubjectUseCase(subjectRepository);
module.exports = {
  subjectRepository,

  getSubjectsUseCase,
  createSubjectUseCase,
};
